go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -322,18 +297,27 @@ func (bsnc *BaseSecondaryNetworkController) addLogicalPortToNetworkForNAD(pod *k
 		return fmt.Errorf("error transacting operations %+v: %v", ops, err)
 	}
 	txOkCallBack()
-	bsnc.podRecorder.AddLSP(pod.UID, bsnc.NetInfo)
 
-	// if somehow lspUUID is empty, there is a bug here with interpreting OVSDB results
-	if len(lsp.UUID) == 0 {
-		return fmt.Errorf("UUID is empty from LSP: %+v", *lsp)
+	if lsp != nil {
+		_ = bsnc.logicalPortCache.add(pod, switchName, nadName, lsp.UUID, podAnnotation.MAC, podAnnotation.IPs)
 	}
 
-	_ = bsnc.logicalPortCache.add(pod, switchName, nadName, lsp.UUID, podAnnotation.MAC, podAnnotation.IPs)
+	// we need to create the binding ourselves for remote ports
```

Comment:
maybe a comment that this is only applicable to layer 2 topologies? or maybe add a check for if the network type is layer2 as well in isRemotePort :=

Reply:
added both